### ISSUE TITLE
Media Manager returning empty results when searching

### ISSUE THREAD

**Issue by igmoweb** [humanmade/altis-enhanced-search#97]
The WP Media Manager returns an empty array when a search is done.

Steps to reproduce (Elastisearch needs to be enabled):
1. Open an existing/create a new post.
2. Insert an image block
3. Click on Media Library
4. Search for anything
5. Results are empty.

This doesn't happen when the same search is done from the Media Library screen or when the search is empty:

- If the search is empty, ElasticPress will disable itself for that search
- A search done by the Media Manager from Gutenberg sends an extra parameter (mime type) and here's where the problem seems to be.

Here's the query I'm triggering in ES (change `"query": "beach" for something that could work for you):

```json
{
  "from": 0,
  "size": 40,
  "sort": [
    {
      "post_date": {
        "order": "desc"
      }
    }
  ],
  "query": {
    "function_score": {
      "query": {
        "bool": {
          "should": [
            {
              "simple_query_string": {
                "query": "beach",
                "fields": [
                  "post_title",
                  "post_excerpt",
                  "post_content",
                  "alt",
                  "rekognition_labels",
                  "attachments.attachment.content"
                ],
                "default_operator": "and"
              }
            }
          ]
        }
      },
      "exp": {
        "post_date_gmt": {
          "scale": "14d",
          "decay": 0.25,
          "offset": "7d"
        }
      },
      "score_mode": "avg",
      "boost_mode": "sum"
    }
  },
  "post_filter": {
    "bool": {
      "must": [
        {
          "terms": {
            "post_mime_type": [
              "image"
            ]
          }
        },
        {
          "term": {
            "post_type.raw": "attachment"
          }
        },
        {
          "terms": {
            "post_status": [
              "inherit",
              "private"
            ]
          }
        }
      ]
    }
  },
  "explain": true
}
```

Note that removing
```json
{
          "terms": {
            "post_mime_type": [
              "image"
            ]
          }
        },
```

would return results with no problem.

Acceptance criteria:

- [ ] Should we filter by mime type? I don't think there's a big reason to filter by type when we're searching. This would only make sense if the user has an available selector to filter by mime type but that's not the case.

**Comment by igmoweb** [comment]
After talking to @roborourke, as the Media Manager sends the Media mime, we need to respect that so this change could solve it:

```
"post_mime_type": [
              "image/jpeg",
              "image/png",
              "image/gif",
            ]
```

**Comment by roborourke** [comment]
Seems that in ElasticPress there's an incorrect assumption that when it receives an array of mime types that they are fully qualified. This is still the case in EP v3.

We'll need to work around this for now but I'll try and fix it upstream too. The offending code is here:

```php
if ( ! empty( $args['post_mime_type'] ) ) {
			if ( is_array( $args['post_mime_type'] ) ) {
				$filter['bool']['must'][] = array(
					'terms' => array(
						'post_mime_type' => (array) $args['post_mime_type'],
					),
				);

				$use_filters = true;
			} elseif ( is_string( $args['post_mime_type'] ) ) {
				$filter['bool']['must'][] = array(
					'regexp' => array(
						'post_mime_type' => $args['post_mime_type'] . '.*',
					),
				);

				$use_filters = true;
			}
		}
```

**Comment by igmoweb** [comment]
@roborourke Worths mentioning that `image/*` won't work either. It needs to specify the types of mime for GIF, JPE and PNG, unless there's another option that I don't know about.

**Comment by roborourke** [comment]
@igmoweb fix released in 4.0.2

**Comment by igmoweb** [comment]
Awesome, that was really fast
